frontend/src/api/client.test.ts

```diff
@@ -0,0 +1,65 @@
+import { AxiosRequestConfig, AxiosResponse } from "axios";
+import { httpClient } from ".";
+
+jest.mock("next/router");
+
+const requestInterceptor = () => {
+  return httpClient.interceptors.request['handlers'][0];
+}
+
+const responseInterceptor = ()=> {
+  return httpClient.interceptors.response['handlers'][0];
+}
+
+const mockRequest: AxiosRequestConfig = {
+  url: undefined,
+  headers: {}
+}
+
+const mockResponse = {
+  response: {
+    status: undefined,
+  }
+}
+
+describe('Client interceptors', () => {
+  describe('Request interceptors', () => {
+    it('should request to an open endpoint without authorization', () => {
+      mockRequest.url = '/users/sign_in';
+      const resolvedReq = requestInterceptor().fulfilled(mockRequest);
+      expect(resolvedReq).toMatchObject(mockRequest);
+      expect('Authorization' in resolvedReq.headers).toBeFalsy();
+    });
+    it('should request closed endpoints with authorization', () => {
+      mockRequest.url = '/users/test';
+      const resolvedReq = requestInterceptor().fulfilled(mockRequest);
+      expect(resolvedReq).toMatchObject(mockRequest);
+      expect('Authorization' in resolvedReq.headers).toBeTruthy();
+    });
+    it('should return error', () => {
+      const mockError = new Error('test');
+      const rejectedReq = requestInterceptor().rejected(mockError);
+      expect(rejectedReq).rejects.toMatchObject<Error>(mockError);
+    });
+  });
+
+  describe('Response interceptors', () => {
+    it('should return correct response when response status is 200', () => {
```

Comment:
I believe that you can remove expression is, correctly @edimossilva?

Reply:
Not sure @iurynogueira , it is easier for me to identify when removing `the`. For me, this looks good